query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -277,16 +273,17 @@ private boolean rhsHasLongValue(Expression rhs) {
    */
   private boolean doesSingleViewFilterMatchLeafQueryFilter(
       Map<String, ViewColumnFilter> viewFilterMap, Filter queryFilter) {
-    if (queryFilter.getLhs().getValueCase() != ValueCase.COLUMNIDENTIFIER
-        && queryFilter.getLhs().getValueCase() != ValueCase.ATTRIBUTE_EXPRESSION) {
+
+    if (queryFilter.getOperator() != Operator.IN && queryFilter.getOperator() != Operator.EQ) {
       return false;
     }
-    if (queryFilter.getOperator() != Operator.IN && queryFilter.getOperator() != Operator.EQ) {
+
+    Optional<String> columnName = getLogicalColumnName(queryFilter.getLhs());
```

Comment:
nit: could be combined with below check, e.g.
```java
ViewColumnFilter viewColumnFilter = columnName.map(viewFilterMap::get).orElse(null);
```

Reply:
resolved